### PR TITLE
MAT-5099 harp id case insensitive

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-5099](https://jira.cms.gov/browse/MAT-5099)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/cqlLibraryList/CqlLibraryList.tsx [hunk 1]
```diff
@@ -253,13 +251,13 @@ export default function CqlLibraryList({ cqlLibraryList, onListUpdate }) {
                 data-testid={`edit-cql-library-button-${selectedCQLLibrary.id}-edit`}
               >
                 {/* edit and version: must be draft and have ownership, else view only*/}
-                {selectedCQLLibrary.createdBy === userName &&
+                {checkUserCanEdit(selectedCQLLibrary.createdBy, []) &&
```
Comment (adongare) [inline src/components/cqlLibraryList/CqlLibraryList.tsx]: can we do it like you are doing it in EditCqlLibrary.tsx? calling this hook once and assigning it to a variable

Comment (sb-cecilialiu) [reply src/components/cqlLibraryList/CqlLibraryList.tsx]: @adongare Thanks a lot Ashok, just made the change.

File: src/components/cqlLibraryList/CqlLibraryList.tsx [hunk 2]
```diff
@@ -142,6 +142,7 @@ export default function CqlLibraryList({ cqlLibraryList, onListUpdate }) {
   const [anchorEl, setAnchorEl] = useState(null);
   const [selectedCQLLibrary, setSelectedCqlLibrary] =
     useState<CqlLibrary>(null);
+  const isOwner = checkUserCanEdit(selectedCQLLibrary?.createdBy, []);
```
Comment (chubert-sb) [inline src/components/cqlLibraryList/CqlLibraryList.tsx:145]: Just a nitpick: I feel like isOwner might not be an appropriate name anymore, but that's something we can adjust later

**Review by chubert-sb — APPROVED** [review]
